### PR TITLE
Add formatting check to CI

### PR DESCRIPTION
**Pull request by blackHatMonkey**
Add yapf check to make sure that all files are always formatted. Since there were not formated a format has been done.

### CODE REVIEW

**Comment by sophron** [comment]
@blackHatMonkey I'm fine with adding this check.

However, you will have to create a page to the wiki "Submitting a Pull Request" where you will have to extensively describe the requirements of any code submitted to the project, including all the standards that we follow as well as short guidelines on how to comply with them (e.g. using `pylint`, `pep8checker` or other tools that you have in mind). We could then point to that page any potential contributors that submitted PRs with failed checks.

**Comment by blackHatMonkey** [comment]
@sophron Can you review this. I have added a wiki page on how to start a pull request and how to test. Let me know if anything needs changing.

**Comment by blackHatMonkey** [comment]
@sophron Can you also make sure that we use the `Github app` version of `Travis-CI`.

**Comment by sophron** [comment]
Yes, I will have a look soon.

I wasn't expecting you to write a tutorial on how to make a Pull Request; there are plenty on the Internet. It's okay that you did it, but what we are more interested about is our specific requirements before merging a PR. In other words, what an experienced open source contributor needs to to know before submitting a PR for Wifiphisher.

**Comment by sophron** [comment]
Imagine someone that knows how to use git, how to submit a PR or how to squash his commits. This contributor is interested to know: 

- What code conventions are we following? PEP8-only?
- How can he check if he follows PEP-8? (lynis example)
- Are there any tools he may use to comply with our coding style (e.g. autopep8?)
- How can the contributor know if his PR "passes" prior to submitting the PR? (usage example of `tox` would be great)

**Comment by blackHatMonkey** [comment]
This is just a first step towards what you are explaining. The is a good set of them [here](https://github.com/mre/awesome-static-analysis#python). We will gradually add these tools to `tox` but the process won't change; All you need is to run `tox`.

**Comment by sophron** [comment]
Brian,

The contributor needs to know about the checks that are part of our CI *before* running `tox`. Before even starts writing code for the project.

I won't merge this PR unless we have documented our CI checks.

**Comment by blackHatMonkey** [comment]
@sophron I have added `Tox tests` section. Let me know if you need me to add anything else in there.

**Comment by blackHatMonkey** [comment]
>Before even starts writing code for the project.

I don't think we need to explicitly state everyone of our coding rules because it's impractical to write and read. Let's work through a PR.
1. user modifies code (doesn't match our standards)
2. runs `tox` and gets errors
3. fixes the errors and goes back to 2
4. submit PR

Now these standard checks are **NOT** implemented in this PR but we are going to add it later.

**Comment by sophron** [comment]
Brian,

What I'm trying to avoid here is a contributor that makes a PR only to realize that we require PEP-8. We need to be clear for this beforehand. We need to have a section in the doc "Before Contributing" where we explicitly state our code policy. The contributor will be able to adjust his `.vimrc` accordingly and he won't be surprised when his PR will fail our checks.

I don't see why it's so difficult to write these now.

**Comment by blackHatMonkey** [comment]
@sophron Would it look something like this:
```
Coding Standards

We require the code to fully comply with the following:
     * PEP8
```

>The contributor will be able to adjust his .vimrc

Do you mean for indentation purposes?

**Comment by sophron** [comment]
Yes, for indendation purposes. But that was an example. The general rule is that the contributor needs to know the code policy before he starts writing code.

It would like this, yes. But you need to add more content. What is PEP8? Is there a tool that automates the code for PEP8, for example?

What other standards are we following apart from PEP-8? How are we formatting our docstrings? If an argument is added, where else does it need to be documented? 

All these need to be answered from this document.

**Comment by blackHatMonkey** [comment]
@sophron That's why I wanted to add it later with pep8 linters but I can add them now if you want?

**Comment by blackHatMonkey** [comment]
Also can you fix the tests failing. I think it failed because you added a feature.

**Comment by sophron** [comment]
Yes, please. We need the documentation first.

**Comment by blackHatMonkey** [comment]
@sophron I have updated it. Let me know if anything needs changing.

File: docs/contribute.rst [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+.. code:: bash
+
+    yapf -i file_name
+
+YAPF will take care of the rest and automatically
+makes the required changes.
+
+It is recommended that you configure your editor
+to automatically format the code using YAPF
+to reduce manual work. For example in `visual studio code`_
```
Comment (sophron) [inline docs/contribute.rst]: I'm curious. How are you using Visual Studio to run a Linux-only tool?

**Comment by sophron** [comment]
Can you split the page into three separate pages?

First would be the `How to Contribute`, another would be `How to Use git` and the last one would be `Code Requirements`.

The first one should be on the index and include links for the other two.

File: docs/contribute.rst (hunk 1 above)
Comment (blackHatMonkey) [reply docs/contribute.rst]: It's [Visual Studio Code](https://code.visualstudio.com/) which is just a cross platform editor.

**Comment by blackHatMonkey** [comment]
@sophron Just made the changes. Let me know if it is good and I can squash it.

**Comment by sophron** [comment]
@blackHatMonkey Since you are writing documentation and it's important for the readers to understand exactly what we are trying to say, I recommend using a grammar checker. I believe "grammarly" offers a free version but there are others too, including "quillbot" (for paraphrasing) and "hemingwayapp".

As soon as you fix syntax and grammar mistakes, I'll continue with adding to your docs and then we can finally merge this.

**Comment by blackHatMonkey** [comment]
@sophron Can you be more specific about the grammar issues?

**Comment by blackHatMonkey** [comment]
@sophron I have made some changes but I'm not sure if these were all the mistakes. Let me know if find others.

File: .travis.yml [hunk 2]
```diff
@@ -5,15 +5,14 @@ language: python
 
 dist: trusty
```
Comment (cclauss) [inline .travis.yml]: Drop trusty because it is EOL.  https://blog.travis-ci.com/2019-04-15-xenial-default-build-environment

File: .travis.yml [hunk 3]
```diff
@@ -5,15 +5,14 @@ language: python
...
-python:
-  - "2.7"
-
 install:
   - pip install pylint tox scapy
   - pip install tornado==4.5.3
   - python setup.py install
 
 script:
+  - pyenv global 2.7.14 3.6.3
```
Comment (cclauss) [inline .travis.yml]: This takes a lot of time so only do it if __python: 2.7__ does not work for your purposes.

**Comment by sophron** [comment]
This is an old PR. Closing due to inactivity.

**Pull request closed without merging**